packages/components-native/src/__mocks__/__mocks.ts

```diff
@@ -22,6 +22,7 @@ jest.mock("react-native-reanimated", () => {
     FadeIn: {
       duration: () => undefined,
     },
+    useEvent: () => ({ listeners: [] }),
```

Comment:
this the the one we actually use in our test suite.

I don't get why we have 2 separate implementations, where we tell consumers to use `jestSetup.js` but we don't use it ourselves? seems strange since the file is right there.

also, I don't think this is the right way to handle this. I was hitting this exact issue of `useEvent` not being found, and there's a clearly recommended way to setup your jest tests.
https://github.com/software-mansion/react-native-reanimated/issues/6335#issuecomment-2308806958

checking in `jobber-mobile` we're doing it via `require('react-native-reanimated').setUpTests();`

and then also using the `jestSetup` from `components-native` but I would bet money that if we removed the reanimated mocks, tests over there would still pass.